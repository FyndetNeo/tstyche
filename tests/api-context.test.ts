import { expect, test } from "@jest/globals";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const fixture = "api-context";

test("includes nested", () => {
  const { status, stderr, stdout } = spawnTyche(fixture, ["nested"]);

  expect(stdout).toMatchSnapshot("stdout");
  expect(stderr).toBe("");

  expect(status).toBe(0);
});

test("context.only", () => {
  const { status, stderr, stdout } = spawnTyche(fixture, ["only"]);

  expect(stdout).toMatchSnapshot("stdout");
  expect(stderr).toBe("");

  expect(status).toBe(0);
});

test("context.skip", () => {
  const { status, stderr, stdout } = spawnTyche(fixture, ["skip"]);

  expect(stdout).toMatchSnapshot("stdout");
  expect(stderr).toBe("");

  expect(status).toBe(0);
});

test("context.todo", () => {
  const { status, stderr, stdout } = spawnTyche(fixture, ["todo"]);

  expect(stdout).toMatchSnapshot("stdout");
  expect(stderr).toBe("");

  expect(status).toBe(0);
});

test("handles 'expect()' nested within 'context()'", () => {
  const { status, stderr, stdout } = spawnTyche(fixture, ["context-level-expect"]);

  expect(stdout).toMatchSnapshot("stdout");
  expect(stderr).toMatchSnapshot("stderr");

  expect(status).toBe(1);
});
