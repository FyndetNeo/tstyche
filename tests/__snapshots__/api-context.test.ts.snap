// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`context.only: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

pass ./__typetests__/context-only.test.ts
  is context?
    - skip is skipped?
  is only context?
    + is never?
    - skip is skipped?
    - todo is todo?
    + is string?
    is nested context?
      + is string?
    is nested only context?
      + is string?
      - todo is todo?
    is nested skipped context?
      - skip is skipped?
      - skip is skipped?
  - skip is skipped?

Targets:    1 passed, 1 total
Test files: 1 passed, 1 total
Tests:      5 skipped, 2 todo, 4 passed, 11 total
Assertions: 2 skipped, 4 passed, 6 total
Duration:   <<timestamp>>

Ran test files matching 'only'.
"
`;

exports[`context.skip: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

pass ./__typetests__/context-skip.test.ts
  is skipped context?
    - skip is skipped?
  is parent context?
    + is string?
    is nested skipped context?
      - skip is skipped too?
      - todo is todo?
    - todo is todo?
  + is string still?

Targets:    1 passed, 1 total
Test files: 1 passed, 1 total
Tests:      2 skipped, 2 todo, 2 passed, 6 total
Assertions: 2 passed, 2 total
Duration:   <<timestamp>>

Ran test files matching 'skip'.
"
`;

exports[`context.todo: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

pass ./__typetests__/context-todo.test.ts
  is todo context?
  is todo context too?
    - todo is todo?
  is parent skipped context?
    - skip is string?
    is nested todo context?
      - todo is todo too?
      - todo is todo too?
      - todo and this is todo?
      - todo is todo?
    - todo is todo?
  + is string still?

Targets:    1 passed, 1 total
Test files: 1 passed, 1 total
Tests:      1 skipped, 6 todo, 1 passed, 8 total
Assertions: 1 passed, 1 total
Duration:   <<timestamp>>

Ran test files matching 'todo'.
"
`;

exports[`handles 'expect()' nested within 'context()': stderr 1`] = `
"Error: 'expect()' cannot be nested within 'context()' helper.

   6 | 
   7 | context("handles nested 'expect()'?", () => {
>  8 |   expect<never>().type.toBeNever();
     |   ^
   9 |   expect<null>().type.toBeNull();
  10 | 
  11 |   test("is number?", () => {

       at ./__typetests__/context-level-expect.tst.ts:8:3

Error: 'expect()' cannot be nested within 'context()' helper.

   7 | context("handles nested 'expect()'?", () => {
   8 |   expect<never>().type.toBeNever();
>  9 |   expect<null>().type.toBeNull();
     |   ^
  10 | 
  11 |   test("is number?", () => {
  12 |     expect<number>().type.toBeNumber();

       at ./__typetests__/context-level-expect.tst.ts:9:3

"
`;

exports[`handles 'expect()' nested within 'context()': stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

fail ./__typetests__/context-level-expect.tst.ts
  + is string?

Targets:    1 failed, 1 total
Test files: 1 failed, 1 total
Tests:      1 passed, 1 total
Assertions: 1 passed, 1 total
Duration:   <<timestamp>>

Ran test files matching 'context-level-expect'.
"
`;

exports[`includes nested: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

pass ./__typetests__/context-nested.tst.ts
  is context?
    + is void?
  is parent context?
    + is never?
    is nested context?
      + is string?
  + is string still?

Targets:    1 passed, 1 total
Test files: 1 passed, 1 total
Tests:      4 passed, 4 total
Assertions: 4 passed, 4 total
Duration:   <<timestamp>>

Ran test files matching 'nested'.
"
`;
